pkg/sources/github/github.go

```diff
@@ -132,173 +141,192 @@ func (s *Source) Init(aCtx context.Context, name string, jobId, sourceId int64,
+	// needed for clones
+	s.token = token
 
-		// needed to list repos
-		ts := oauth2.StaticTokenSource(
-			&oauth2.Token{AccessToken: cred.Token},
-		)
-		tc := oauth2.NewClient(context.TODO(), ts)
+	// needed to list repos
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(context.TODO(), ts)
 
-		var err error
-		// If we're using public github, make a regular client.
-		// Otherwise make an enterprise client
-		if apiEndpoint == "https://api.github.com" {
-			apiClient = github.NewClient(tc)
-		} else {
-			apiClient, err = github.NewEnterpriseClient(apiEndpoint, apiEndpoint, tc)
-			if err != nil {
-				return errors.New(err)
-			}
+	var err error
+	// If we're using public github, make a regular client.
+	// Otherwise make an enterprise client
+	var apiClient *github.Client
+	if apiEndpoint == "https://api.github.com" {
+		apiClient = github.NewClient(tc)
+	} else {
+		apiClient, err = github.NewEnterpriseClient(apiEndpoint, apiEndpoint, tc)
+		if err != nil {
+			return nil, errors.New(err)
 		}
+	}
 
-		// TODO: this should support scanning users too
+	// TODO: this should support scanning users too
 
-		specificScope := false
+	specificScope := false
 
-		if len(s.repos) > 0 {
-			specificScope = true
-		}
+	if len(s.repos) > 0 {
+		specificScope = true
+	}
 
-		if len(s.orgs) > 0 {
-			specificScope = true
-			for _, org := range s.orgs {
-				errOrg := s.addReposByOrg(ctx, apiClient, org)
-				errUser := s.addReposByUser(ctx, apiClient, org)
-				if errOrg != nil && errUser != nil {
-					log.WithError(errOrg).Error("error fetching repos for org or user: ", org)
-				}
+	if len(s.orgs) > 0 {
+		specificScope = true
+		for _, org := range s.orgs {
+			errOrg := s.addReposByOrg(ctx, apiClient, org)
+			errUser := s.addReposByUser(ctx, apiClient, org)
+			if errOrg != nil || errUser != nil {
+				log.WithError(errOrg).Error("error fetching repos for org or user: ", org)
 			}
 		}
+	}
 
-		user, _, err := apiClient.Users.Get(context.TODO(), "")
-		if err != nil {
-			return errors.New(err)
-		}
-
-		if !specificScope {
-			s.addReposByUser(ctx, apiClient, user.GetLogin())
-			// Scan for orgs is default with a token. GitHub App enumerates the repositories
-			// that were assigned to it in GitHub App settings.
-			s.addOrgsByUser(ctx, apiClient, user.GetLogin())
-			for _, org := range s.orgs {
-				s.addReposByOrg(ctx, apiClient, org)
-			}
-		}
+	user, _, err := apiClient.Users.Get(context.TODO(), "")
+	if err != nil {
+		return nil, errors.New(err)
+	}
 
-		s.addGistsByUser(ctx, apiClient, user.GetLogin())
+	// If no scope was provided, enumerate them
+	if !specificScope {
+		s.addReposByUser(ctx, apiClient, user.GetLogin())
+		// Scan for orgs is default with a token. GitHub App enumerates the repositories
+		// that were assigned to it in GitHub App settings.
+		s.addOrgsByUser(ctx, apiClient, user.GetLogin())
```

Comment:
We should probably heed the linter here, check these errors and log appropriately.